tensorflow/core/kernels/stateful_random_ops_gpu.cu.cc

```diff
@@ -26,12 +26,13 @@ limitations under the License.
 #include "tensorflow/core/util/gpu_kernel_helper.h"
 #include "tensorflow/core/util/gpu_launch_config.h"
 
+// ROCm hipMemcpyToSymbol can only see this variable if it's in global namespace 
```

Comment:
@ekuznetsov139 

Can you break this PR into the following two functional commits
1. change to move the thread counter variable into the global namespace
2. change to pack the functor args into a struct

It will help make the upstream PR (i.e. the PR we will need to file to upstream this change into the official TF repo) easy to review and approve. :)